source/common/filter/tcp_proxy.cc

```diff
@@ -441,13 +447,18 @@ void TcpProxy::onUpstreamEvent(Network::ConnectionEvent event) {
     connect_timeout_timer_.reset();
   }
 
-  if (event == Network::ConnectionEvent::RemoteClose ||
-      event == Network::ConnectionEvent::LocalClose) {
+  // upstream_connection_ can be nullptr when we received a RemoteClose, called into
+  // closeUpstreamConnection() in this block, and are now receiving a LocalClose event.
+  if ((upstream_connection_ != nullptr) && (event == Network::ConnectionEvent::RemoteClose ||
+                                            event == Network::ConnectionEvent::LocalClose)) {
     disableIdleTimer();
```

Comment:
Is it expected to disable the idle timer before tcp proxy session teardown?  I just want to sanity check if we get a connection close while draining this change won't cause problems.

Reply:
That's a good point. I think it's unrelated to this change, but the timer probably shouldn't be disabled until the downstream side has completed FlushWrite. I'll need to think about that a little more and stare at the code a bit.